proto/tsopb.proto

```diff
@@ -0,0 +1,128 @@
+syntax = "proto3";
+package tsopb;
+
+import "gogoproto/gogo.proto";
+import "rustproto.proto";
+
+option (gogoproto.sizer_all) = true;
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+option (rustproto.lite_runtime_all) = true;
+
+option java_package = "org.tikv.kvproto";
+
+service TSO {
+    rpc Tso(stream TsoRequest) returns (stream TsoResponse) {}
+
+    rpc SyncMaxTS(SyncMaxTSRequest) returns (SyncMaxTSResponse) {}
+
+    rpc GetDCLocationInfo(GetDCLocationInfoRequest) returns (GetDCLocationInfoResponse) {}
+
+    rpc SetExternalTimestamp(SetExternalTimestampRequest) returns (SetExternalTimestampResponse) {}
+
+    rpc GetExternalTimestamp(GetExternalTimestampRequest) returns (GetExternalTimestampResponse) {}
+}
+
+// TenantID is the unique id of the tenant
+message TenantID {
+    // id is a sequntial id. 
+    uint32 id = 1;
+    // keyspace_group_id is the unique id of the keyspace to which the tenant belongs
+    uint32 keyspace_group_id = 2;
+}
+
+message RequestHeader {
+    // tenant_id is the unique id of the tenant
+    TenantID tenant_id = 1;
```

Comment:
How about keyspace_id?

Reply:
Didn't notice there is cluster_id in the RequestHeader/ResponseHeader. In that case, let me flatten the struct.

Reply:
changed to
`message RequestHeader {
    // tenant_id is the unique id of the tenant
    uint32 tenant_id = 1;
    // keyspace_group_id is the unique id of the keyspace to which the tenant belongs
    uint32 keyspace_group_id = 2;
    // cluster_id is the ID of the cluster which be sent to.
    uint64 cluster_id = 3;
    // sender_id is the ID of the sender server, also member ID or etcd ID.
    uint64 sender_id = 4;
}

message ResponseHeader {
    // tenant_id is the unique id of the tenant
    uint32 tenant_id = 1;
    // keyspace_group_id is the unique id of the keyspace to which the tenant belongs
    uint32 keyspace_group_id = 2;
    // cluster_id is the ID of the cluster which sent the response.
    uint64 cluster_id = 3;
    Error error = 4;
}`